tests/core/test_rule_groups.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from smdebug_rulesconfig.debugger_rules._constants import (
+    SUPPORTED_DL_FRAMEWORKS,
+    SUPPORTED_FRAMEWORKS,
+)
+from smdebug_rulesconfig.debugger_rules._ruleGroups import (
+    DEEP_LEARNING_APPLICATION_RULES,
+    DEEP_LEARNING_RULES,
+    UNIVERSAL_RULES,
+    XGBOOST_RULES,
+)
+from smdebug_rulesconfig.debugger_rules._utils import _get_rule_list
+
+
+@pytest.mark.parametrize(
+    "rule_list, framework_list",
+    [
+        (UNIVERSAL_RULES, SUPPORTED_FRAMEWORKS),
+        (DEEP_LEARNING_RULES, SUPPORTED_DL_FRAMEWORKS),
+        (DEEP_LEARNING_APPLICATION_RULES, SUPPORTED_DL_FRAMEWORKS),
+        (XGBOOST_RULES, ["xgboost"]),
+    ],
+)
+def test_framework_rule_compatibility(rule_list, framework_list):
+    for rule in rule_list:
+        for framework in framework_list:
+            assert rule in _get_rule_list(framework), f"Rule: {rule} not supported by {framework}"
+
+
+@pytest.mark.parametrize(
+    "rule_list, framework_list",
+    [
+        (DEEP_LEARNING_RULES, SUPPORTED_DL_FRAMEWORKS + ["xgboost"]),
+        (XGBOOST_RULES, ["xgboost"] + SUPPORTED_DL_FRAMEWORKS),
+    ],
+)
+def test_framework_rule_incompatibility(rule_list, framework_list):
+    for rule in rule_list:
+        for framework in framework_list:
+            if rule not in _get_rule_list(framework):
+                if rule in DEEP_LEARNING_RULES:
+                    assert framework == "xgboost"
+                elif rule in XGBOOST_RULES:
+                    assert framework in SUPPORTED_DL_FRAMEWORKS
```

Comment:
I think we can make this test more simple. Basically, we just need to affirm that no debugger rules should be in the xgboost rule list and no xgboost rules should be in any DL framework's rule list. So the test cases could be like:

(deep learning rules, [xgboost])
(xgboost rules, supported dl frameworks)

and the test could be like:

```
for framework in framework_list:
    assert not (set(rule_list) & set(_get_rule_list(framework)))
```

Reply:
(if we do this, ignore the previous comment about the fixtures)